hadoop-hdds/docs/content/design/storage-distribution.md

```diff
@@ -132,6 +132,19 @@ Additionally, Recon already possesses a comprehensive physical and logical capac
 A new SCM upgrade action (ScmOnFinalizeActionForDataDistribution) is introduced. 
 This action is part of the finalization process for the DATA_DISTRIBUTION layout feature, which enables the new block size tracking capabilities.
 
+A new feature layout is added to check compatibility between components. Following is an example for handling compatability between OM and SCM
+
+| OM Version | SCM Version | Compatibility Handling                                                                                                                    |
+|------------|-------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| **Old OM** | **New SCM** | SCM receives old proto. In `ServerSideTranslatorPB`, it checks if the new block list is empty; if so, it decodes using the old structure. |
+| **New OM** | **Old SCM** | OM will be using `getScmInfo()` to fetch SCM metadata layout version. If the feature is not finalized, OM sends the old structure.        |
+| **New OM** | **New SCM** | Fully upgraded. OM sends size-aware transactions and SCM processes them accordingly.                                                      |
+| **Old OM** | **Old SCM** | Legacy setup. Both components use old proto structure.                                                                                    |
```

Comment:
We currently don't support rolling upgrades so OM/SCM cross compatibility does not need to be handled like this. `ComponentVersion`s are currently used for client/server interactions and `LayoutFature`s are used to keep disk state readable to an older version when downgrading.